src/main/java/com/twilio/http/Request.java

```diff
@@ -56,9 +66,15 @@ public void setAuth(final String username, final String password) {
      * @return basic authentication string
      */
     public String getAuthString() {
-        final String credentials = this.username + ":" + this.password;
-        final String encoded = Base64.getEncoder().encodeToString(credentials.getBytes(StandardCharsets.US_ASCII));
-        return "Basic " + encoded;
+        if (username != null && password != null) {
```

Comment:
Can we convert username and password into basic auth strategy in any of the previous steps and just follow auth strategy based methods here?

Reply:
That is a good suggestion, I think we might have to check if it will break existing customer who are using below configuration:
https://github.com/twilio/twilio-java/blob/main/advanced-examples/custom-http-client.md